cmd/config.go

```diff
@@ -20,17 +20,17 @@ var configCmd = &cobra.Command{
 	Long: `With this command you can configure your settings. Start off by running
 "bonusly config --token <your-api-token>" to be able to use the Bonusly CLI.
 
- If you don't have a Bonusly API token yet, go visit https://bonus.ly/api to create one.`,
-
+If you don't have a Bonusly API token yet, go visit https://bonus.ly/api to create one.`,
 	Run: func(cmd *cobra.Command, args []string) {
-		if len(args) == 0 {
+		if cmd.Flag("token").Name == "token" && cmd.Flag("token").Changed {
```

Comment:
I'm sorry, I don't think I understand this part fully.
What is the advantage of doing all that instead of just `len(apiToken) == 0`?
`cmd.Flag("token").Name == "token"... etc.` seems like a lot of additional unneeded code to me. I'm not saying this is bad - this might very well be the proper way to do this kind of thing. I just lack the knowledge of the cobra library to understand what exactly this code is doing.

Reply:
I have never used cobra before. According to my tests `bonusly config` leads to the execution of configCmd.Run(). In this context args only count non-flag arguments. It is 0 even issuing `bonusly config something else`. This makes the test _if len(args) == 0_ useless.
So I start searching when the token flag should be handled and now reading your comment I realized I overshoot  the code. I've made some simplifications to correct this.

In resum the logic is:

Is there a token flag?
	Is flag value empty?
		Value can't be empty
	Is flag value invalid? (not implemented)
		Value should ve valig (not implemented)
Please, provide a flag

Reply:
Sounds good!